implementation/src/main/java/io/smallrye/config/EnvConfigSource.java

```diff
@@ -253,16 +260,46 @@ static Optional<List<Integer>> indexOfDashes(final String property, final int of
                     matchPosition--;
                 }
             } else if (c != envProperty.charAt(matchPosition)) {
-                return Optional.empty();
+                return MINUS_ONE.get();
             } else {
                 matchPosition--;
             }
         }
         if (matchPosition >= eoffset) {
-            return Optional.empty();
+            return MINUS_ONE.get();
+        }
+
+        return dashesPosition.get();
+    }
+
+    static class ResizableIntArray {
+        static final ResizableIntArray ZERO = with(0);
+        static final ResizableIntArray MINUS_ONE = with(-1);
+
+        private int[] array;
+
+        ResizableIntArray(int initialSize) {
+            this.array = new int[initialSize];
         }
 
-        return Optional.of(dashesPosition);
+        int[] get() {
+            return array;
+        }
+
+        void ensureCapacity(int capacity) {
```

Comment:
enlarge only if needed, not always!

Reply:
I don't reuse the array between calls, and when `ensureCapacity` is called, it always requires increasing the size.

Reply:
Yes, the most common case scenario is to exit with `0` or `-1`. Also, I need two methods calls to check the prefix and the rest of the name. I could use two separate arrays instances and juggle these around, but I think it is not worth it.